Source/src/ui/ImguiUtils.cpp

```diff
@@ -86,6 +86,45 @@ bool Hachiko::ImGuiUtils::CollapsingHeader(GameObject* game_object, Component* c
     return open;
 }
 
+bool Hachiko::ImGuiUtils::ToolbarButton(ImFont* font, const char* font_icon, bool active, const char* tooltip_desc)
+{
+    const ImVec4 col_active = ImGui::GetStyle().Colors[ImGuiCol_ButtonActive];
+    const ImVec4 bg_color = active ? col_active : ImGui::GetStyle().Colors[ImGuiCol_Text];
+
+    ImGui::SameLine();
+    const auto frame_padding = ImGui::GetStyle().FramePadding;
+    ImGui::PushStyleColor(ImGuiCol_FrameBg, ImVec4(0, 0, 0, 0));
+    ImGui::PushStyleColor(ImGuiCol_Text, bg_color);
+    ImGui::PushStyleColor(ImGuiCol_Button, ImVec4(0, 0, 0, 0));
+    ImGui::PushStyleColor(ImGuiCol_ButtonActive, ImVec4(0, 0, 0, 0));
+    ImGui::PushStyleVar(ImGuiStyleVar_FramePadding, ImVec2(0, ImGui::GetStyle().FramePadding.y));
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, frame_padding);
+    ImGui::PushStyleVar(ImGuiStyleVar_FrameRounding, 0);
+
+    ImGui::PushFont(font);
+    active = ImGui::Button(font_icon);
+    ImGui::PushStyleColor(ImGuiCol_Text, ImVec4(1, 1, 1, 1));
+    ImGui::SameLine(); DisplayTooltip(tooltip_desc);
+
+    ImGui::PopFont();
+    ImGui::PopStyleColor(5);
+    ImGui::PopStyleVar(3);
+
+    return active;
+}
+
+static void Hachiko::ImGuiUtils::DisplayTooltip(const char* desc)
+{
+    if (ImGui::IsItemHovered())
+    {
+        ImGui::BeginTooltip();
+        ImGui::PushTextWrapPos(ImGui::GetFontSize() * 35.0f);
```

Comment:
I think this `35.0f` can be a `constexpr` defined in `ImguiUtils` namespace that explains what it is.

Reply:
Yes, this would certainly make it clearer. Added!

Reply:
Thank you!!